EIPS/eip-7231.md

```diff
@@ -0,0 +1,175 @@
+Exisiting solutions such as ENS, although open, decentralized, and make it more convenient for Ethereum-based applications, lack of data standardization and athentication of itentity due to inherent anominity. Other solutions such as SBTs rely on centralized attestors, can not prevent data tampering, nor inscribe data into the ledger itself in a privacy enabling way. Previous work such as [ERC-6066](./eip-6066.md) does not bind nor verify user identity data, thus unable to meet the user identity aggregation, user Kyc verification and other application scenarios related to user data sovereignty
+The proposed pushes the boundaries of solving identity problems with Identity Aggregated NFT, i.e., the individual-authenticated aggregation of web2 and web3 identities to NFTs (SBTs included). 
+
+## Specification
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY” and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+### Every compliant contract must implement the Interface
+
+```typescript
+// SPDX-License-Identifier: CC0-1.0
+pragma solidity ^0.8.15;
+
+interface IERC7231 {
+
+    /**
+     * @notice emit the use binding informain
+     * @param id nft id 
+     * @param identitiesRoot new identity root
+     */
+    event SetIdentitiesRoot(
+        uint256 id,
+        bytes32 identitiesRoot
+    );
+
+    /**
+     * @notice 
+     * @dev set the user ID binding information of NFT with identitiesRoot
+     * @param id nft id 
+     * @param identitiesRoot multi UserID Root data hash
+     * MUST allow external calls
+     */
+    function setIdentitiesRoot(
+        uint256 id,
+        bytes32 identitiesRoot
+    ) external;
+
+    /**
+     * @notice 
+     * @dev get the multi-userID root by  NFTID
+     * @param id nft id 
+     * MUST return the bytes32 multiUserIDsRoot
+     * MUST NOT modify the state
+     * MUST allow external calls
+     */
+    function getIdentitiesRoot(
+        uint256 id
+    ) external returns(bytes32);
+
+    /**
+     * @notice 
+     * @dev verify the userIDs binding 
+    * @param id nft id 
+     * @param userIDs userIDs for check
+     * @param identitiesRoot msg hash to veriry
+     * @param signature ECDSA signature 
+     * MUST If the verification is passed, return true, otherwise return false
+     * MUST NOT modify the state
+     * MUST allow external calls
+     */
+    function verifyIdentitiesBinding(
+        uint256 id,address nftOwnerAddress,string[] memory userIDs,bytes32 identitiesRoot, bytes calldata signature
+    ) external returns (bool);
+}
+```
+
+This is the “ERCXXX Metadata JSON Schema” referenced above.
```

Comment:
I don't see where this is mentioned above?

Reply:
Thanks，has been corrected